prow/cmd/jenkins-operator/logs.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/NYTimes/gziphandler"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/jenkins"
+)
+
+// serveLogs starts a http server and serves Jenkins logs.
+// Meant to be called inside a goroutine.
+func serveLogs(jc *jenkins.Client) {
+	http.Handle("/", gziphandler.GzipHandler(handleLog(jc)))
+	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+}
+
+// Matches letters, numbers, hyphens, and underscores.
+var objReg = regexp.MustCompile(`^[\w-]+$`)
+
+func handleLog(jc *jenkins.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		// Needs to be a GET request.
+		if r.Method != http.MethodGet {
+			http.Error(w, "Forbidden.", http.StatusForbidden)
+			return
+		}
+
+		// Needs to get Jenkins logs.
+		if !strings.HasSuffix(r.URL.Path, "consoleText") {
+			http.Error(w, "Forbidden.", http.StatusForbidden)
+			return
+		}
+
+		log, err := jc.Get(r.URL.Path)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Log not found: %v", err), http.StatusNotFound)
+			logrus.WithError(err).Warning("Error returned.")
```

Comment:
Please include more context in this error message. Specifically, that the error is a result of trying to fetch logs from jenkins.

Reply:
done